File: src/main/java/factorymethod/SortCreator.java
package factorymethod;

public class SortCreator {

    static ISort selectionSort() {
        return new SelectionSort();
    }

    static ISort heapsort() {
        return new Heapsort();
    }

    static ISort radixSort() {
        return new RadixSort();
    }
}

File: src/main/java/generative_patterns/abstractFactory_factoryMethod/SortCreator.kt
package generative_patterns.abstractFactory_factoryMethod

object SortCreator {
    fun selectionSort(): ISort {
        return SelectionSort()
    }

    fun heapsort(): ISort {
        return Heapsort()
    }

    fun radixSort(): ISort {
        return RadixSort()
    }
}